internal/services/firewall/firewall_policy_rule_collection_group_resource.go

```diff
@@ -65,6 +65,7 @@ func resourceFirewallPolicyRuleCollectionGroup() *pluginsdk.Resource {
 			"application_rule_collection": {
 				Type:     pluginsdk.TypeList,
 				Optional: true,
+				Computed: true,
```

Comment:
The provider is now in a trend to avoid `Computed` + `Optional`. So this change shall be not expected. Instead, to make the configuration valid, users are expected to add an explicit `ignore_changes` for this block.